### PR TITLE
Sort items and filter targets without packets

### LINKED ISSUE #1172: Sort items in TGT/PKT/ITEM dropdown alphabetically
Currently the ITEMS in the target packet item drop down is sorted by bit offset. This is not intuitive since the targets and packets are sorted alphabetically.

### PR DESCRIPTION
**Pull request by jmthomas**
closes #1172

### CODE REVIEW

File: openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/TargetPacketItemChooser.vue [hunk 1]
```diff
@@ -242,14 +242,24 @@ export default {
...
         if (target == 'UNKNOWN') {
           return []
         }
         return { label: target, value: target }
       })
+      // Filter out any targets without packets
+      for (var i = this.targetNames.length - 1; i >= 0; i--) {
+        const cmd =
+          this.mode === 'tlm' ? 'get_all_tlm_names' : 'get_all_cmd_names'
+        await this.api[cmd](this.targetNames[i].value).then((names) => {
```
Comment (jmthomas) [inline openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/TargetPacketItemChooser.vue:257]: I don't love having to request the names but it's in the created() method and it's the only way to determine if the target has packets.

File: openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/TargetPacketItemChooser.vue [hunk 2]
```diff
@@ -350,9 +360,13 @@ export default {
         (packet) => {
           this.itemNames = packet.items
             .map((item) => {
+              let label = item.name
+              if (item.data_type == 'DERIVED') {
+                label += ' *'
+              }
```
Comment (jmthomas) [inline openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/TargetPacketItemChooser.vue:366]: Add a * just like in PacketViewer for DERIVED items

File: openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/TargetPacketItemChooser.vue (hunk 1 above)
Comment (ryanmelt) [reply openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/TargetPacketItemChooser.vue:257]: This is too heavy with lots of API calls for lots of targets.

Comment (ryanmelt) [reply openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/TargetPacketItemChooser.vue:257]: Make this a todo and note we can't do lots of API calls.

**Review by ryanmelt — CHANGES_REQUESTED** [review]

**Pull request merged**